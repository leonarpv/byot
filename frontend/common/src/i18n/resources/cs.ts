/**
 * Plural keys
 * 0 - singular
 * 1 - 2,3,4
 * 2 - 0 and >5
 */
export default {
  'Login': 'Přihlášení',
  'Username or email': 'Použivatelské jméno nebo email',
  'Enter username or email': 'Zadejte použivatelské jméno nebo email',
  'Enter password': 'Zadejte heslo',
  'Password': 'Heslo',
  'Forgot password?': 'Zapomenuté heslo',
  "Don't have an account?": 'Vytvořit konto',
  'Sign up': 'Registrace',
  'Enter valid email': 'Zadejte platný email',
  'Enter email': 'Zadejte email',
  'Password must be longer than 8 characters': 'Heslo musí mít alespoň 8 znaků',
  'Password must contain at least one special character': 'Heslo musí obsahovat alespoň jeden speciální znak',
  'Confirm password': 'Potvrzení hesla',
  'Passwords are not the same': 'Heslá se neshodují',
  'You have to agree with our terms and conditions': 'Musíte souhlasit se všeobecnými podmínkami',
  'First name': 'Jméno',
  'Last name': 'Přimění',
  'Email': 'Email',
  'I allow this website to collect and store submitted data.': 'Souhlasím se spracovaním osobních údajů.',
  'Already have an account?': 'Již máte konto?',
  'Log in': 'Přihlaste se',
  'Request reset password': 'Požádat o změnu hesla',
  'Reset password': 'Zabudnuté heslo',
  'All rights reserved to': 'Všechna práva vyhrazena pro',
  'Back': 'Zpět',
  'Error while trying to login, please, check username or password':
    'Nastala chyba při přihlašovaní, skontrolujte, prosím, zadané údaje',
  'Log out': 'Odhlásit se',
  'training_0': 'trénink',
  'training_1': 'tréninky',
  'training_2': 'tréninků',
  'video_0': 'video',
  'video_1': 'videa',
  'video_2': 'videí',
  'Enter training name': 'Název tréninku',
  'Add video': 'Přidat video',
  'Save training': 'Uložit trénink',
  'Video label': 'Název videa',
  'Start typing to search or paste URL': 'Vyhledejte nebo vložte URL',
  'My training sets': 'Moje seznamy tréninkov',
  'Nothing found': 'Nic nenalezeno',
  'Type to search': 'Pro vyhledávaní začněte psát',
  'Search': 'Vyhledávaní',
  'Home': 'Domů',
  'Create training set': 'Vytvořit seznam tréninků',
  'My profile': 'Můj profil',
};
