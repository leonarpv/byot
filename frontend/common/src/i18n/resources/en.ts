export default {
  'Login': 'Login',
  'Username or email': 'Username or email',
  'Enter username or email': 'Enter username or email',
  'Enter password': 'Enter password',
  'Password': 'Password',
  'Forgot password?': 'Forgot password?',
  "Don't have an account?": "Don't have an account?",
};
