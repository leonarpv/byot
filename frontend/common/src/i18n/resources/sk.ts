export default {
  'Login': 'Prihlásenie',
  'Username or email': 'Používateľské meno alebo email',
  'Enter username or email': 'Zadajte používateľské meno alebo email',
  'Enter password': 'Zadajte heslo',
  'Password': 'Heslo',
  'Forgot password?': 'Zabudnuté heslo',
  "Don't have an account?": 'Vytvoriť konto',
  'Sign up': 'Registrácia',
  'Enter valid email': 'Zadajte platný email',
  'Enter email': 'Zadajte email',
  'Password must be longer than 8 characters': 'Heslo musí mať aspoň 8 znakov',
  'Password must contain at least one special character': 'Heslo musí obsahovať aspoň jeden špeciálny znak',
  'Confirm password': 'Potvrdenie hesla',
  'Passwords are not the same': 'Heslá sa nezhodujú',
  'You have to agree with our terms and conditions': 'Musíte súhlasiť s všeobecnými podmienkami',
  'First name': 'Meno',
  'Last name': 'Priezvisko',
  'Email': 'Email',
  'I allow this website to collect and store submitted data.': 'Súhlasím so spracovaním osobných údajov.',
};
