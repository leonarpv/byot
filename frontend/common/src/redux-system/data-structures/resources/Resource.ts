export enum ResourceState {
  IDLE = 'IDLE',
  LOADING = 'LOADING',
}

export interface Resource<T> {
  data?: T | undefined;
  isProcessing: boolean;
  isLoaded: boolean;
  hasData: boolean;
  state: ResourceState;
}
