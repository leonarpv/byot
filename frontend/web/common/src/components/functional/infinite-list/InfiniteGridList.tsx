import * as React from 'react';
import {InfiniteList, InfiniteListProps} from './InfiniteList';
import {GridProps, Grid} from '@material-ui/core';

interface Props<T> extends InfiniteListProps<T> {
  GridProps?: GridProps;
}

export const InfiniteGridList = <T extends any>(props: Props<T>) => {
  const [className, setClassName] = React.useState('');
  const ref = React.useCallback((node: HTMLDivElement) => setClassName(node.className), [props.GridProps]);
  return (
    <>
      <div style={{display: 'none'}}>
        <Grid ref={ref} {...props.GridProps} />
      </div>
      <InfiniteList {...props} className={className} />
    </>
  );
};
