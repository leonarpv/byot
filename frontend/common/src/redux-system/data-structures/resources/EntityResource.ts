import {Resource, ResourceState} from './Resource';
import * as _ from 'lodash';

export class EntityResource<T> implements Resource<T> {
  protected _data?: T;
  protected _isLoaded: boolean = false;

  /**
   * @param defaultData is supposed to be just simple structure or primitive (eg. {}, [], null, 0)
   * @param state
   */
  constructor(protected defaultData?: T, public state: ResourceState = ResourceState.IDLE) {
    this._data = _.cloneDeep(this.defaultData);
  }

  public get hasData(): boolean {
    return !!this._data;
  }

  public get isProcessing(): boolean {
    return [ResourceState.LOADING].includes(this.state);
  }

  get data(): T | undefined {
    return this._data;
  }

  setData<D = T | undefined>(value: D) {
    this._isLoaded = true;
    this._data = (value as unknown) as T | undefined;
  }

  get isLoaded(): boolean {
    return this._isLoaded;
  }

  public reset<R extends Resource<T> = EntityResource<T>>(): R {
    return (new EntityResource<T>(this.defaultData) as unknown) as R;
  }
}
