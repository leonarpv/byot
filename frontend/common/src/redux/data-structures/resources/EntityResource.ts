import { Resource, ResourceState } from './Resource';
import * as _ from 'lodash';

export class EntityResource<T> implements Resource<T> {
  protected _data?: T;
  protected _isLoaded: boolean = false;

  constructor(
    private defaultData?: T,
    public state: ResourceState = ResourceState.IDLE
  ) {
    this._data = _.cloneDeep(this.defaultData);
  }

  public get hasData(): boolean {
    return !!this._data;
  }

  public get isProcessing(): boolean {
    return [ResourceState.LOADING].includes(this.state);
  }

  get data(): T | undefined {
    return this._data;
  }

  set data(value: T | undefined) {
    this._isLoaded = true;
    this._data = value;
  }

  get isLoaded(): boolean {
    return this._isLoaded;
  }
}
