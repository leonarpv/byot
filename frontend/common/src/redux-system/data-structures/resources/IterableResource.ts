import {EntityResource} from './EntityResource';
import {ResourceState} from './Resource';
import {IList} from '../../../types/interfaces/IList';

export class IterableResource<T> extends EntityResource<T[]> {
  public totalCount: number;
  public appendNewData: boolean;

  constructor(
    defaultData: T[] = [],
    {
      state = ResourceState.IDLE,
      totalCount = defaultData.length,
      appendNewData = true,
    }: {
      state?: ResourceState;
      totalCount?: number;
      appendNewData?: boolean;
    } = {}
  ) {
    super(defaultData, state);
    this.totalCount = totalCount;
    this.appendNewData = appendNewData;
  }

  get hasData(): boolean {
    return this.data!.length > 0;
  }

  set data(value: T[] | IList) {
    const data = Array.isArray(value) ? value || [] : value.entries;
    this._data = this.appendNewData ? [...this._data, ...data] : data;
    this.totalCount = Array.isArray(value) ? value.length : value.meta?.totalCount;
  }

  get data(): T[] {
    return this._data || [];
  }

  reset() {
    super.reset();
    this.totalCount = this.data.length;
  }
}
