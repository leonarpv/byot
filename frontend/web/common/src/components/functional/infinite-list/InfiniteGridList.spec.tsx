import * as React from 'react';
import {render} from '@testing-library/react';
import {InfiniteGridList} from './InfiniteGridList';
import {IterableResource} from '@byot-frontend/common/src/redux-system/data-structures/resources/IterableResource';

describe('<InfiniteGridList/>', () => {
  it('should render', () => {
    const {container} = render(
      <InfiniteGridList next={jest.fn()} resource={new IterableResource<any>([])} />
    );
    expect(container).toMatchSnapshot();
  });
});
