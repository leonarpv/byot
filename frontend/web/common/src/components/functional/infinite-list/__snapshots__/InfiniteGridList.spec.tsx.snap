// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfiniteGridList/> should render 1`] = `
<div>
  <div
    style="display: none;"
  >
    <div
      class="MuiGrid-root"
    />
  </div>
  <div
    class="infinite-scroll-component__outerdiv"
  >
    <div
      class="infinite-scroll-component MuiGrid-root"
      style="height: auto; overflow: auto;"
    >
      <div
        class="makeStyles-root-1"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-2"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
